pretyping/pretyping.ml

```diff
@@ -436,7 +436,15 @@ let pretype_ref ?loc sigma env ref us =
   match ref with
   | GlobRef.VarRef id ->
       (* Section variable *)
-      (try sigma, make_judge (mkVar id) (NamedDecl.get_type (lookup_named id !!env))
+    (try
+       let ty = NamedDecl.get_type (lookup_named id !!env) in
+       (match us with
+        | None | Some [] -> ()
```

Comment:
Why don't you forbid writing an instance at all? AFAIK `@{}` still means something even for polymorphic constants.

Reply:
`@{}` ensures no universes are generated imo, which is valid for variables too

Reply:
This is true when *declaring* a variable, but not when *interpreting* it. I believe we should fail in this piece of code.

Reply:
>This is true when declaring a variable, but not when interpreting it

?? Seems true when interpreting too (after this PR). 

I don't see why we would fail.

Reply:
This piece of code drops the instance, so I don't understand what is the point of allowing `@{}` when we know this is irrelevant. What I agree with is that we should be able to write `Let x@{} := ...` to ensure that no universe constraints are generated.

Reply:
It doesn't drop the instance, it checks that it has the correct size.

Reply:
... and drops it afterwards, which means that it was irrelevant.

Reply:
I don't know what you mean by drop tbh

Reply:
Note that I don't think this is blocking the PR, I just find this introduces a discrepancy between the user-facing representation and what actually exists at the level of kernel terms.